runtime/sharedstate_test.go

```diff
@@ -0,0 +1,259 @@
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+	. "github.com/onflow/cadence/runtime/tests/utils"
+)
+
+func TestRuntimeSharedState(t *testing.T) {
+
+	t.Parallel()
+
+	runtime := newTestInterpreterRuntime()
+
+	signerAddress := common.MustBytesToAddress([]byte{0x1})
+
+	deploy1 := DeploymentTransaction("C1", []byte(`
+        pub contract C1 {
+            pub fun hello() {
+                log("Hello from C1!")
+            }
+        }
+    `))
+
+	deploy2 := DeploymentTransaction("C2", []byte(`
+        pub contract C2 {
+            pub fun hello() {
+                log("Hello from C2!")
+            }
+        }
+    `))
+
+	accountCodes := map[common.Location][]byte{}
+
+	var events []cadence.Event
+	var loggedMessages []string
+
+	var interpreterState *interpreter.SharedState
+
+	var ledgerReads []ownerKeyPair
+
+	ledger := newTestLedger(
+		func(owner, key, value []byte) {
+			ledgerReads = append(
+				ledgerReads,
+				ownerKeyPair{
+					owner: owner,
+					key:   key,
+				},
+			)
+		},
+		nil,
+	)
+
+	runtimeInterface := &testRuntimeInterface{
+		storage: ledger,
+		getSigningAccounts: func() ([]Address, error) {
+			return []Address{signerAddress}, nil
+		},
+		updateAccountContractCode: func(address Address, name string, code []byte) error {
+			location := common.AddressLocation{
+				Address: address,
+				Name:    name,
+			}
+			accountCodes[location] = code
+			return nil
+		},
+		getAccountContractCode: func(address Address, name string) (code []byte, err error) {
+			location := common.AddressLocation{
+				Address: address,
+				Name:    name,
+			}
+			code = accountCodes[location]
+			return code, nil
+		},
+		removeAccountContractCode: func(address Address, name string) error {
+			location := common.AddressLocation{
+				Address: address,
+				Name:    name,
+			}
+			delete(accountCodes, location)
+			return nil
+		},
+		resolveLocation: func(identifiers []ast.Identifier, location common.Location) (result []sema.ResolvedLocation, err error) {
+
+			// Resolve each identifier as an address location
+
+			for _, identifier := range identifiers {
+				result = append(result, sema.ResolvedLocation{
+					Location: common.AddressLocation{
+						Address: location.(common.AddressLocation).Address,
+						Name:    identifier.Identifier,
+					},
+					Identifiers: []ast.Identifier{
+						identifier,
+					},
+				})
+			}
+
+			return
+		},
+		log: func(message string) {
+			loggedMessages = append(loggedMessages, message)
+		},
+		emitEvent: func(event cadence.Event) error {
+			events = append(events, event)
+			return nil
+		},
+		setInterpreterSharedState: func(state *interpreter.SharedState) {
+			interpreterState = state
+		},
+		getInterpreterShareState: func() *interpreter.SharedState {
```

Comment:
```suggestion
		getInterpreterSharedState: func() *interpreter.SharedState {
```

Reply:
Fixed in 3d44ae4a28b25764b4d9b146e25db6f8d280f6a4